File: rb/features.rb
# Motivational SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MotivationalFeatures
  def self.make_feature(name)
    case name
    when "base"
      MotivationalBaseFeature.new
    when "test"
      MotivationalTestFeature.new
    else
      MotivationalBaseFeature.new
    end
  end
end

File: py/motivational_sdk/features.py
# Motivational SDK feature factory

from motivational_sdk.feature.base_feature import MotivationalBaseFeature
from motivational_sdk.feature.test_feature import MotivationalTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MotivationalBaseFeature(),
        "test": lambda: MotivationalTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
